src/decode.rs

```diff
@@ -47,8 +47,24 @@ impl Event {
         self.id = value.to_vec();
     }
 
-    fn to_dispatch_event(&mut self) -> () {
-        self.data.truncate(self.data.len() - 1);
+    fn to_dispatch_event(self) -> Option<Self> {
```

Comment:
Add some rules to the spec. These are some of the todo's I had noted on the Event struct.